tests/test_job.py

```diff
@@ -97,15 +88,47 @@ def list_namespaced_pod_error(namespace, label_selector):
             "start_time": START_TIME- datetime.timedelta(minutes=1)}}
     return MockObject({"items":[MockObject(return_value)]})
 
+def list_namespaced_pod_pending(namespace, label_selector):
+    return_value =  {"status": {"conditions": [],
+            "container_statuses": [MockObject({"container_id": None,
+                                    "image": "ubuntu_mock_test_image",
+                                    "image_id": "",
+                                    "last_state": {"running": None,
+                                                   "terminated": None,
+                                                   "waiting": None},
+                                    "name": "task-1000-ex-00",
+                                    "ready": False,
+                                    "restart_count": 0,
+                                    "state": {"running": None,
+                                              "terminated": None,
+                                              "waiting": {"message": "Unknown error",
+                                                          "reason": "Unknown"}}})],
+            "host_ip": "192.168.99.100",
+            "init_container_statuses": None,
+            "message": None,
+            "nominated_node_name": None,
+            "phase": "Pending",
+            "pod_ip": "172.17.0.5",
+            "qos_class": "BestEffort",
+            "reason": None,
+            "start_time": START_TIME- datetime.timedelta(minutes=5)}}
+    return MockObject({"items":[MockObject(return_value)]})
+
```

Comment:
Make life easier to the reviewer and come up with more descriptive names. What use case are you testing? The Pod is Pending - but why? Is it ImagePullError, but too early? Is it imagePullError and it should stop the execution? Or maybe a different reason?

The new tests are : Running and Pending with a different cause. Minimally there is the one missing that presents "the image is wrong, but it is too early to stop."